src/main/user-impl/java/com/mysql/cj/jdbc/ha/MultiHostConnectionProxy.java

```diff
@@ -330,18 +326,17 @@ protected void invalidateCurrentConnection() throws SQLException {
      * @throws SQLException
      *             if an error occurs
      */
-    protected void invalidateConnection(JdbcConnection conn) throws SQLException {
-        synchronized (this.lockObject) {
-            try {
-                if (conn != null && !conn.isClosed()) {
-                    conn.realClose(true, !conn.getAutoCommit(), true, null);
-                }
-            } catch (SQLException e) {
-                // swallow this exception, current connection should be useless anyway.
+    protected synchronized void invalidateConnection(JdbcConnection conn) throws SQLException {
```

Comment:
We can remove synchronized here. InvalidateConnection() has synchronized so no need to do it twice.